bot/xchange/Cryptopia.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+require_once __DIR__ . '/../CCXTAdapter.php';
+
+define( 'CRYPTOPIA_ID', 6 );
+
+class CryptopiaExchange extends \ccxt\cryptopia {
+
+  use CCXTErrorHandler;
+
+  public function nonce() {
+    return generateNonce( CRYPTOPIA_ID );
+  }
+};
+
+class Cryptopia extends CCXTAdapter {
+
+  public function __construct() {
+    parent::__construct( CRYPTOPIA_ID, 'Cryptopia', 'CryptopiaExchange' );
+  }
+
+  public function getRateLimit() {
+      $limit = $this->exchange->rateLimit;
+      return $limit;
+  }
+
+  public function isMarketActive( $market ) {
+    return $market[ 'active' ];
+  }
+
+  public function checkAPIReturnValue( $result ) {
+//    if ( !$result[ 'info' ][ 'Error' ] ) {
+//      return false;
+//    }
+    return $result[ 'info' ][ 'Success' ] === true;
+  }
+
+  public function getDepositHistory() {
+
+      $history = [ ];
+
+      $result = $this->exchange->privatePostGetTransactions( [ 'Type' => 'Deposit' ] );
+
+      if($result[ 'Success' ]) {
+          $history = $result[ 'Data' ];
+      }
+
+    return array(
+
+      'history' => $history,
+      'statusKey' => 'Status',
+      'coinKey' => 'Currency',
+      'amountKey' => 'Amount',
+      'timeKey' => 'TimeStamp',
+      'pending' => 'Pending',
+      'completed' => 'Success',
+
+    );
+
+  }
+
+  public function getWithdrawalHistory() {
+
+      $history = [ ];
+
+      $result = $this->exchange->privatePostGetTransactions( [ 'Type' => 'Deposit' ] );
+
+      if($result[ 'Success' ]) {
+          $history = $result[ 'Data' ];
+      }
+
+    return array(
+
+      'history' => $history,
+      'statusKey' => 'Status',
+      'coinKey' => 'Currency',
+      'amountKey' => 'Amount',
+      'timeKey' => 'TimeStamp',
+      'txidKey' => 'TxId',
+      'addressKey' => 'Address',
+      'pending' => 'Pending',
+      'completed' => 'Success',
+
+    );
+
+  }
+
+    // Changed functions!
```

Comment:
Why have you made these changes?

Reply:
your adapter not universal! and have specific code